modules/setting/config_provider.go

```diff
@@ -66,41 +65,49 @@ func newConfigProviderFromData(configContent string) (ConfigProvider, error) {
 	}, nil
 }
 
+type Options struct {
+	CustomConf                string // the ini file path
+	AllowEmpty                bool   // whether not finding configuration files is allowed (only true for the tests)
+	ExtraConfig               string
+	IgnoreCheckRunUser        bool
+	DisableLoadCommonSettings bool
+}
+
 // newConfigProviderFromFile load configuration from file.
 // NOTE: do not print any log except error.
-func newConfigProviderFromFile(customConf string, allowEmpty bool, extraConfig string) (*iniFileConfigProvider, error) {
+func newConfigProviderFromFile(opts *Options) (*iniFileConfigProvider, error) {
 	cfg := ini.Empty()
 	newFile := true
 
-	if customConf != "" {
-		isFile, err := util.IsFile(customConf)
+	if opts.CustomConf != "" {
+		isFile, err := util.IsFile(opts.CustomConf)
 		if err != nil {
-			return nil, fmt.Errorf("unable to check if %s is a file. Error: %v", customConf, err)
+			return nil, fmt.Errorf("unable to check if %s is a file. Error: %v", opts.CustomConf, err)
 		}
 		if isFile {
-			if err := cfg.Append(customConf); err != nil {
-				return nil, fmt.Errorf("failed to load custom conf '%s': %v", customConf, err)
+			if err := cfg.Append(opts.CustomConf); err != nil {
+				return nil, fmt.Errorf("failed to load custom conf '%s': %v", opts.CustomConf, err)
 			}
 			newFile = false
 		}
 	}
 
-	if newFile && !allowEmpty {
+	if newFile && !opts.AllowEmpty {
 		return nil, fmt.Errorf("unable to find configuration file: %q, please ensure you are running in the correct environment or set the correct configuration file with -c", CustomConf)
 	}
 
-	if extraConfig != "" {
-		if err := cfg.Append([]byte(extraConfig)); err != nil {
+	if opts.ExtraConfig != "" {
+		if err := cfg.Append([]byte(opts.ExtraConfig)); err != nil {
 			return nil, fmt.Errorf("unable to append more config: %v", err)
 		}
 	}
 
 	cfg.NameMapper = ini.SnackCase
+	ignoreCheckRunUser = opts.IgnoreCheckRunUser
```

Comment:
It seems still fragile. I would suggest do not do so.

Refactor the logger system instead of patching it.

Reply:
Done.